### DISCUSSION TITLE
Serving static files with resteasy-servlet-spring-boot-starter in springboot #4301

### DISCUSSION THREAD

**Discussion by liweinan** [resteasy/resteasy-spring-boot#340] [Q&A] [answered]
> [srpsingh](https://github.com/srpsingh)
> [6 hours ago](https://github.com/orgs/resteasy/discussions/4301#discussion-7110259)
I'm using resteasy-servlet-spring-boot-starter 6.2.0.Final with springboot 3.2.2. I'm not sure how can we serve static files from /META-INF/resources/api. https://docs.jboss.org/resteasy/docs/6.2.9.Final/userguide/ suggests that we can use filter org.jboss.resteasy.plugins.server.servlet.FilterDispatcher but it doesn't seem to help as the servlet HttpServlet30Dispatcher just sends not-found response back. Looking for pointers to resolve this.

**Comment by liweinan** [comment]
@srpsingh I'll find time to check your question and give back my answer by the end of next week.

**Reply by srpsingh** [reply]
ok, thanks.

**Reply by liweinan** [reply]
@srpsingh Here is a simple solution: https://stackoverflow.com/a/51074375/1212922

Does it solve your problem?

**Reply by liweinan** [reply]
I haven't checked whether this solution works or not: 

- https://www.baeldung.com/spring-mvc-static-resources

If I have time I'll verify the above solution works or not.

**Reply by liweinan** [reply, marked as answer]
I have verified the above solution works:

- https://github.com/liweinan/Resteasy2168SpringBoot/pull/5

**Comment by liweinan** [comment]
I assume this question is answered. If there is any more problem please reopen the discussion.

**Reply by srpsingh** [reply]
@liweinan  Thanks. I was able to use your information and find a solution. I was much fixated on Filter way of resolving it. I've put the following code (need a lot of improvements though) that seem to be working:

@Component
@Path("")
public class StaticResourcesResource
{

    @Inject
    ServletContext context;

    @GET
    //    @Path("{path: ^api\\/.*}")
    @Path("{path: ^api.*}")
    public Response staticResources(@PathParam("path") final String path)
    {

        // FIXME there must be better way to make this response
        String processedPath = null;
        if (path.equals("api") || path.equals("api/"))
        {
            processedPath = "api/index.html";
        }
        else
        {
            processedPath = path;
        }

        InputStream resource = context.getResourceAsStream(String.format("%s", processedPath));

        String contentType = null;
        if (processedPath.endsWith(".html"))
        {
            contentType = "text/html";
        } else if (processedPath.endsWith(".js"))
        {
            contentType = "application/javascript";
        }
        else if (processedPath.endsWith(".css"))
        {
            contentType = "text/css";
        }
        else if (processedPath.endsWith(".png"))
        {
            contentType = "image/png";
        }

        return Objects.isNull(resource)
                ? Response.status(NOT_FOUND).build()
                : Response.ok().entity(resource).type(contentType).build();
    }
}
